### ISSUE TITLE
Guest checkout for downloadable product results in no links being sent in confirmation email

### ISSUE THREAD

**Issue by crankycyclops** [magento/magento2#20028]
Environment:

Tested on Magento 2.2.7 and PHP 7.1.

Summary of issue:

If you place an order for a downloadable product as a guest, the confirmation email contains no links. I enabled shareable downloads both globally and for the individual products that were tested. 

Steps to reproduce:

1. Open browser in incognito mode
2. Add a downloadable product to the cart and checkout as a guest
3. Check the confirmation email

Expected result:

Confirmation email will contain links for the downloadable product.

Actual result:

Confirmation email has no links.

Other notes:

I tested this with a downloadable product that contains more than one file.

**Comment by magento-engcom-team** [comment]
Hi @crankycyclops. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@crankycyclops do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by crankycyclops** [comment]
@magento-engcom-team give me 2.2.7 instance

**Comment by magento-engcom-team** [comment]
Hi @crankycyclops. Thank you for your request. I'm working on Magento 2.2.7 instance for you

**Comment by magento-engcom-team** [comment]
Hi @crankycyclops, here is your [Magento instance](https://i-20028-2-2-7.instances.magento-community.engineering).
Admin access: [https://i-20028-2-2-7.instances.magento-community.engineering/admin](https://i-20028-2-2-7.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by shikhamis11** [comment]
@crankycyclops please correct the issue create format
https://github.com/magento/magento2/wiki/Issue-reporting-guidelines

**Comment by crankycyclops** [comment]
@shikhamis11 it actually looks like this is a duplicate of #19036 and #18323, so I'll go ahead and close this. I'll make sure to follow the guidelines the next time I report an issue.
